public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -117,6 +137,42 @@ export class TempoDatasource extends DataSourceWithBackend<TempoQuery, TempoJson
     }
   }
 
+  init = async () => {
+    const response = await lastValueFrom(
+      this._request('/api/status/buildinfo').pipe(
+        map((response) => response),
+        catchError((error) => {
+          console.error('Failure in retrieving build information', error.data.message);
+          return of({ error, data: { version: null } }); // unknown version
+        })
+      )
+    );
+    this.tempoVersion = response.data.version;
+  };
+
+  /**
+   * Check, for the given feature, whether it is available in Grafana.
+   *
+   * The check is done based on the version of the Tempo instance running on the backend and
+   * the minimum version required by the given feature to work.
+   *
+   * @param featureName - the name of the feature to consider
+   * @return true if the feature is available, false otherwise
+   */
+  private isFeatureAvailable(featureName: FeatureName) {
+    // We know for old Tempo instances we don't know their version, so resort to default
+    const actualVersion = this.tempoVersion ?? defaultTempoVersion;
+
+    try {
+      return semver.gte(actualVersion, featuresToTempoVersion[featureName]);
+    } catch {
+      // An error could still happen if Tempo is running locally. In that case, the version is not a
+      // semantic version and instead it is in the form `<branch>-<revision>` (e.g., `main-12bdeff`).
+      console.error(`Cannot compare ${actualVersion} and ${featuresToTempoVersion[featureName]}`);
```

Comment:
This is what I currently see when I am running via our local devenv with `traceQLStreaming` enabled and Tempo `2.2.0`. If that's the case, what's the best way to use the streaming feature locally in the frontend?